src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -184,28 +184,28 @@ private SecurityStatusPal DecryptData()
             return PrivateDecryptData(_internalBuffer, ref _decryptedBytesOffset, ref _decryptedBytesCount);
         }
 
-        private SecurityStatusPal PrivateDecryptData(byte[] buffer, ref int offset, ref int count)
+        private SecurityStatusPal PrivateDecryptData(byte[]? buffer, ref int offset, ref int count)
         {
-            return _context.Decrypt(buffer, ref offset, ref count);
+            return _context!.Decrypt(buffer, ref offset, ref count);
         }
 
         //
         // This method assumes that a SSPI context is already in a good shape.
         // For example it is either a fresh context or already authenticated context that needs renegotiation.
         //
-        private Task ProcessAuthentication(bool isAsync = false, bool isApm = false, CancellationToken cancellationToken = default)
+        private Task? ProcessAuthentication(bool isAsync = false, bool isApm = false, CancellationToken cancellationToken = default)
         {
-            Task result = null;
+            Task? result = null;
 
             ThrowIfExceptional();
 
             if (isAsync)
             {
-                result = ForceAuthenticationAsync(new AsyncSslIOAdapter(this, cancellationToken), _context.IsServer, null, isApm);
+                result = ForceAuthenticationAsync(new AsyncSslIOAdapter(this, cancellationToken), _context!.IsServer, null, isApm);
             }
             else
             {
-                ForceAuthenticationAsync(new SyncSslIOAdapter(this), _context.IsServer, null).GetAwaiter().GetResult();
+                ForceAuthenticationAsync(new SyncSslIOAdapter(this), _context!.IsServer, null).GetAwaiter().GetResult();
```

Comment:
This `Task?` result permeates all the way up to the public API. I would recommend changing so that `Task.CompletedTask` is returned in the synchronous branch.

Unless intentional, we should additionally also make this method async and await the authentication method.

Reply:
> This Task? result permeates all the way up to the public API.

What do you mean?  The result Task is only ever used when isAsync is true.

Reply:
> Unless intentional, we should additionally also make this method async and await the authentication method.

This is intentional.  This method should not be made async.

Reply:
> What do you mean? The result Task is only ever used when isAsync is true. This is not observable by callers of the public method.

You're right. I suppose though we could change line 198 to `Task result = Task.CompletedTask;` in order to avoid having to make nullability assertions based on input arguments.

Reply:
Addressed in https://github.com/dotnet/runtime/pull/32541/commits/a613646eb341dc4e0b1158a294f86a706d35eda6